core/src/main/clojure/xtdb/sql/plan.clj

```diff
@@ -138,6 +138,79 @@
                             {::err/message (str "Cannot build interval for: "  (pr-str qualifier))
                              :qualifier qualifier}))))
 
+(defn ->date-diff-call [t1 t2 qualifier]
+  (r/zmatch qualifier
+            [:interval_qualifier
+             [:single_datetime_field [:non_second_primary_datetime_field start-field]]]
+            ;; =>
+            (list 'date_diff t1 t2 start-field)
+  
+            [:interval_qualifier
+             [:single_datetime_field [:non_second_primary_datetime_field start-field] [:unsigned_integer _leading-precision]]]
```

Comment:
Discarding "leading precision" here - seems we do not use this within our intervals either?

Reply:
You're probably right, but I'd prefer we spend some time to become more confident about it :slightly_smiling_face: